docs/features/software-catalog/external-integrations.md

```diff
@@ -1031,3 +1031,360 @@ backend.add(catalogModuleCustomDataParser);
+
+This approach has the following benefits,
+
+1. Reduced ingestion latency - each burst commits entities which are processed before the entire list is processed.
+2. Stable pressure - each period between bursts provides an opportunity for the processing pipeline to settle without overwhelming the pipeline with a large number of unprocessed entities.
+3. Built-in retry / back-off - Failed bursts are automatically retried with a built-in back-off interval providing an opportunity for the data source to reset its rate limits before retrying the burst.
+4. Prevents orphan entities - Deleted entities are removed as with `full` mutation with a low memory footprint.
+
+### Requirements
+
+The Incremental Entity Provider backend is designed for data sources that provide paginated results. Each burst attempts to handle one or more pages of the query. The plugin will attempt to fetch as many pages as it can within a configurable burst length. At every iteration, it expects to receive the next cursor that will be used to query in the next iteration. Each iteration may happen on a different replica. This has several consequences:
+
+1. The cursor must be serializable to JSON (not an issue for most RESTful or GraphQL based APIs).
+2. The client must be stateless - a client is created from scratch for each iteration to allow distributing processing over multiple replicas.
+3. There must be sufficient storage in Postgres to handle the additional data. (Presumably, this is also true of sqlite, but it has only been tested with Postgres.)
+
+### Installation
+
+1. Install `@backstage/plugin-catalog-backend-module-incremental-ingestion` with `yarn --cwd packages/backend add @backstage/plugin-catalog-backend-module-incremental-ingestion` from the Backstage root directory.
+
+2. Add the following code to the `packages/backend/src/index.ts` file:
+
+```ts title="packages/backend/src/index.ts"
+/* highlight-add-next-line */
+import { catalogModuleCustomIncrementalIngestionProvider } from './extensions/catalogCustomIncrementalIngestion';
+
+const backend = createBackend();
+
+/* highlight-add-start */
+backend.add(
+  import(
+    '@backstage/plugin-catalog-backend-module-incremental-ingestion/alpha'
+  ),
+);
+/* highlight-add-end */
+
+// We have created this in section **Adding an Incremental Entity Provider to the catalog**
+/* highlight-add-next-line */
+backend.add(catalogModuleCustomIncrementalIngestionProvider);
+
+backend.start();
+```
+
+### Writing an Incremental Entity Provider
+
+To create an Incremental Entity Provider, you need to know how to retrieve a single page of the data that you wish to ingest into the Backstage catalog. If the API has pagination and you know how to make a paginated request to that API, you'll be able to implement an Incremental Entity Provider for this API. For more information about compatibility, check out the [requirements](#requirements) section of this page.
+
+Here is the type definition for an Incremental Entity Provider.
```

Comment:
Mmm maybe don't put the full source code of that in here. It'll just turn out to drift from reality when we iterate on code. Sometimes it's better to have a little bit of plain text saying something like "Incremental providers need to have a unique name, and some methods that move the cursor and fetch a page of data." and then when the actual implementation starts below, it'll be clear what you are doing in those terms.